blaze/expr/tests/test_reductions.py

```diff
@@ -97,3 +97,33 @@ def test_max_min_on_datetime_and_timedelta(reduc, measure):
 
 def test_reduction_naming_with_generated_leaves():
     assert symbol('_', 'var * float64').sum()._name == 'sum'
+
+
+@pytest.mark.parametrize(
+    ['reduction', 'group_by', 'sort_by', 'preceding', 'following'],
+    list(product(
+        ['sum', 'mean', 'count'],
+        list('ab'),
+        list('ab'),
+        [None, 0, 3],
+        [None, 0, 3]
+    ))
+)
+def test_basic_over(reduction, group_by, sort_by, preceding, following):
+    t = symbol('t', 'var * {a: int64, b: ?float64}')
+    expr = getattr(t.b, reduction)().over(
+        group_by=group_by,
+        sort_by=sort_by,
```

Comment:
similar to what @kwmsmith said we could allow something like the following:

```python
df.a.sum().over(preceding=2).sort(df.c).group_by(df.b)
df.a.sum().over(preceding=2).group_by(df.b).sort(df.c)
df.a.sum().over(preceding=2).sort(df.c)
df.a.sum().over(preceding=2).group_by(df.b)
```

We *could* do this by having some hidden classes like `WindowGroupBy`, `WindowSortBy` whose methods simply create a new `Window` instance when you call their `sort` or `group_by` methods